test/check-application

```diff
@@ -447,20 +466,84 @@ class TestNavigator(testlib.MachineCase):
+
         # Delete button text should match item type: directory/file
         b.mouse("[data-item='newfile']", "contextmenu")
-        b.wait_in_text(".context-menu li:nth-child(8) button", "Delete")
+        b.wait_in_text(".context-menu li:nth-child(10) button", "Delete")
         b.click(".context-menu button:contains('Delete')")
         b.click("button.pf-m-danger")
         b.wait_not_present("[data-item='newfile']")
 
+    def testDownloadUser(self):
+        b = self.browser
+        m = self.machine
+
+        # For group ownership test
+        m.execute("""
+            groupadd navigator-test
+            gpasswd -a admin navigator-test
+        """)
+
+        self.enter_navigator(False)
+        self.allowDownload()
+
+        # create file as root in the users homedir, should be able to read
+        m.execute("echo 'test content' > /home/admin/newfile")
+        size = int(m.execute("stat --format '%s' /home/admin/newfile").strip())
+
+        b.mouse("[data-item='newfile']", "contextmenu")
+        b.wait_in_text(".context-menu li:nth-child(8) button", "Download")
+        b.click(".context-menu button:contains('Download')")
+        self.waitDownloadFile("newfile", size, "test content")
+
+        m.execute("chmod 600 /home/admin/newfile")
+        b.click("[data-item='newfile']")
+        b.wait_text("#description-list-owner-permissions dd", "Read and write")
+        b.wait_text("#description-list-group-permissions dd", "None")
+        b.wait_text("#description-list-other-permissions dd", "None")
+        b.mouse("[data-item='newfile']", "contextmenu")
+        b.wait_not_present(".context-menu button:contains('Download')")
+
+        # group ownership test
+        m.execute("""
+            echo 'test content' > /home/admin/groupfile
+            chown :navigator-test /home/admin/groupfile
+            chmod 640 /home/admin/groupfile
+        """)
+        size = int(m.execute("stat --format '%s' /home/admin/groupfile").strip())
+
+        b.mouse("[data-item='groupfile']", "contextmenu")
+        b.wait_text("#description-list-owner-permissions dd", "Read and write")
+        b.wait_text("#description-list-group-permissions dd", "Read-only")
+        b.wait_text("#description-list-other-permissions dd", "None")
+        b.wait_in_text(".context-menu li:nth-child(8) button", "Download")
+        b.click(".context-menu button:contains('Download')")
+        self.waitDownloadFile("groupfile", size, "test content")
+
+        m.execute("chmod g-r /home/admin/groupfile")
+        b.wait_text("#description-list-owner-permissions dd", "Read and write")
+        b.wait_text("#description-list-group-permissions dd", "None")
+        b.wait_text("#description-list-other-permissions dd", "None")
+        b.mouse("[data-item='groupfile']", "contextmenu")
+        b.wait_not_present(".context-menu button:contains('Download')")
+
+        # Big file downloads fine
+        m.execute("truncate -s 500M /home/admin/test.iso")
```

Comment:
500 MiB still fits into the VM's RAM. How long does this take? Could this be 1.5 GiB to be guaranteed to not fit into RAM and thus get read in chunks?